morgoth/xpi.py

```diff
@@ -57,8 +59,21 @@ class BadXPIfile(Exception):
         pass
 
     def __init__(self, path):
-        if not os.path.isfile(path):
-            raise XPI.DoesNotExist()
+        if path.startswith("https://") or path.startswith("http://"):
+            response = requests.get(path)
```

Comment:
We could add a retry here, e.g. [redo](https://pypi.org/project/redo/) if that makes sense to do.